src/BenchmarkDotNet/Portability/Memory/SysctlMemoryInfoProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using BenchmarkDotNet.Helpers;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Portability.Memory
+{
+    /// <summary>    
+    /// Memory information from output of the `sysctl -a` command.
+    /// MacOSX only.
+    /// </summary>
+    internal static class SysctlMemoryInfoProvider
+    {
+        internal static readonly Lazy<MemoryInfo> SysctlMemoryInfo = new Lazy<MemoryInfo>(Load);
+
+        [CanBeNull]
+        private static MemoryInfo Load()
+        {
+            if (RuntimeInformation.IsMacOSX())
+            {
+                string sysctlContent = ProcessHelper.RunAndReadOutput("sysctl", "-a");
```

Comment:
We already call `sysctl` in `SysctlCpuInfoProvider`. Such calls are pretty expensive. Please, reuse existed data from `SysctlCpuInfoProvider` instead of an additional call.

Reply:
Right, added code to reuse call output for CPU and memory